src/openrct2/CmdlineSprite.cpp

```diff
@@ -452,6 +457,157 @@ int32_t cmdline_for_sprite(const char** argv, int32_t argc)
+        if (loadedObject == nullptr)
+        {
+            fprintf(stderr, "Unable to load object.\n");
+            return -1;
+        }
+        auto entryIndex = object_manager_get_loaded_object_entry_index(loadedObject);
+        uint8_t objectType = entry->GetType();
+
+        auto& objManager = context->GetObjectManager();
+        auto metaObject = objManager.GetLoadedObject(objectType, entryIndex);
+
+        char outputPath[MAX_PATH];
+        safe_strcpy(outputPath, argv[2], MAX_PATH);
+        path_end_with_separator(outputPath, MAX_PATH);
+
+        if (!platform_ensure_directory_exists(outputPath))
+        {
+            fprintf(stderr, "Unable to create directory.\n");
+            return -1;
+        }
+
+        int32_t maxIndex = static_cast<int32_t>(metaObject->GetNumImages());
+        int32_t imagesOffset = 0;
+        switch (objectType)
+        {
+            case OBJECT_TYPE_RIDE:
+            {
+                auto rideEntry = get_ride_entry(entryIndex);
+                imagesOffset = rideEntry->images_offset;
+                break;
+            }
+            case OBJECT_TYPE_SMALL_SCENERY:
+            case OBJECT_TYPE_LARGE_SCENERY:
+            case OBJECT_TYPE_WALLS:
+            case OBJECT_TYPE_BANNERS:
+            case OBJECT_TYPE_PATH_BITS:
+            {
+                auto& objMgr = OpenRCT2::GetContext()->GetObjectManager();
+                auto obj = objMgr.GetLoadedObject(objectType, entryIndex);
+                if (obj != nullptr)
+                {
+                    auto sceneryEntry = static_cast<rct_scenery_entry*>(obj->GetLegacyData());
+                    imagesOffset = sceneryEntry->image;
+                }
+                break;
+            }
+            case OBJECT_TYPE_PATHS:
+            {
+                auto pathEntry = get_path_surface_entry(entryIndex);
+                imagesOffset = pathEntry->image;
+                break;
+            }
+            case OBJECT_TYPE_SCENERY_GROUP:
+            {
+                auto sceneryGroupEntry = get_scenery_group_entry(entryIndex);
+                imagesOffset = sceneryGroupEntry->image;
+                break;
+            }
+            case OBJECT_TYPE_PARK_ENTRANCE:
+            {
+                auto& objMgr = OpenRCT2::GetContext()->GetObjectManager();
+                auto obj = objMgr.GetLoadedObject(objectType, entryIndex);
+                if (obj != nullptr)
+                {
+                    auto entranceEnty = static_cast<rct_entrance_type*>(obj->GetLegacyData());
+                    imagesOffset = entranceEnty->image_id;
+                }
+                break;
+            }
+            default:
+            {
+                fprintf(stderr, "Cannot extract images from this type of object.\n");
+                return -1;
+            }
+        }
+
+        int32_t numbers = std::max(1, static_cast<int32_t>(std::floor(std::log(maxIndex))));
```

Comment:
`numbers` seems a bit generic

Reply:
It was copied from the other function. It saves how many digits there are in the total number of images. Perhaps `numDigits` is better?

Reply:
Does look better, IMHO